src/main/java/io/confluent/connect/jdbc/sink/JdbcDbWriter.java

```diff
@@ -79,15 +82,44 @@ void closeQuietly() {
     cachedConnectionProvider.close();
   }
 
-  TableId destinationTable(String topic) {
-    final String tableName = config.tableNameFormat.replace("${topic}", topic);
+  TableId destinationTable(SinkRecord record) {
+    StringBuilder name = new StringBuilder();
+    final String schemaName = destinationSchema(record);
+    if (!schemaName.isEmpty()) {
+      name.append(schemaName).append(".");
+    }
+    name.append(config.tableNameFormat.replace("${topic}", record.topic()));
+
+    final String tableName = name.toString();
+
     if (tableName.isEmpty()) {
       throw new ConnectException(String.format(
           "Destination table name for topic '%s' is empty using the format string '%s'",
-          topic,
+          record.topic(),
           config.tableNameFormat
       ));
     }
     return dbDialect.parseTableIdentifier(tableName);
   }
+
+  String destinationSchema(SinkRecord record) {
+    StringBuilder schemaName = new StringBuilder();
+    String schemaNameFormat = config.schemaNameFormat;
+    if (!schemaNameFormat.isEmpty() && (record.key() instanceof Struct)) {
+      Struct keyData = ((Struct) record.key());
+      Pattern pattern = Pattern.compile("\\$\\{(.*?)\\}");
```

Comment:
This is a relatively expensive operation. Could you make this pattern into a constant? E.g. `INLINE_VARIABLE_PATTERN`?